### PR TITLE
jQuery function to transition width/height changes.

### PR DESCRIPTION
**Pull request by daniel-qcode**

### CODE REVIEW

**Comment by daniel-qcode** [comment]
http://codepen.io/daniel-qcode/pen/QNJagW

**Comment by bvw** [comment]
Looks good. Handing over to Peter for comments.

File: js/jquery.transition.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        // Default option values
+        options.duration = options.duration || '0.3s';
+        options.timingFunction = options.timingFunction || 'ease';
+
+        $(this).each(function() {
+            var autoValue;
+            var $element = $(this);
+
+            // remove any esiting transitionend.transition events bound to this element.
```
Comment (PeterChaplin) [inline js/jquery.transition.js]: Typo "esiting"

File: js/jquery.transition.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    // ======================================================================
+    $.fn.transition = function(options) {
+        // options.property (required) eg. 'height' or 'width'
+        // options.endValue (required) eg. 'auto', '' (initial value) or '100px' (absolute value)
+        // options.duration (optional) eg. '0.3s', '300ms' etc.
+        // options.timingFunction (optional) eg 'linear', 'ease' etc.
+
+        // Default option values
+        options.duration = options.duration || '0.3s';
+        options.timingFunction = options.timingFunction || 'ease';
```
Comment (PeterChaplin) [inline js/jquery.transition.js]: As a general coding pattern, I think I would prefer:
```js
options = $.extend({
  duration: '0.3s',
  timingFunction: 'ease'
}, options);
```

File: js/jquery.transition.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        $(this).each(function() {
```
Comment (PeterChaplin) [inline js/jquery.transition.js]: Should be no need to wrap `this` here.

**Comment by daniel-qcode** [comment]
Updated

**Pull request merged**